common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -371,13 +388,186 @@ def get_child_descriptors(self):
         return list(self._get_selected_child_blocks())
 
 
+@XBlock.wants('user')
 @XBlock.wants('library_tools')  # Only needed in studio
-class AdaptiveLibraryContentModule(AdaptiveLibraryContentFields, LibraryContentModule):
+class AdaptiveLibraryContentModule(AdaptiveLibraryContentFields, LibraryContentModule, AdaptiveLearningAPIMixin):
     """
     A specialized version of Randomized Content Block that communicates
     with an external service to determine when to show its children, and which ones.
     """
 
+    @lazy
+    def parent_course(self):
+        """
+        Return parent course.
+        """
+        return self.runtime.modulestore.get_course(self.course_id, depth=1)
+
+    @lazy
+    def parent_unit_id(self):
+        """
+        Return block ID of parent unit.
+        """
+        parent_unit = self.get_parent()
+        return parent_unit.scope_ids.usage_id.block_id
+
+    @lazy
+    def anonymous_user_id(self):
+        """
+        Return anonymous ID for current user.
+
+        Note that we can't use `user_service.get_anonymous_user_id` here
+        because it only produces meaningful results for staff users.
```

Comment:
FYI From the [discussion on TNL-836|https://openedx.atlassian.net/browse/TNL-836?focusedCommentId=89275&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-89275], it sounds like edX are worried about the security implications of allowing arbitrary users to access anyone's anonymous Id.  So they locked it down to only allow logged-in staff users to get a meaningful result from the function.

However, I bet it would be ok to allow arbitrary users to access their _own_ anonymous ID, but not anyone else's.  If you think it's likely that edX would accept this change, then I'd recommend using it into this branch too, to avoid any changes in anonymous IDs stored in Domoscio.